public/pages/CreateDetector/containers/CreateDetector.tsx

```diff
@@ -252,6 +252,27 @@ export default class CreateDetector extends Component<CreateDetectorProps, Creat
     });
   };
 
+  getDetectorWithUpdatedRules(newRules: RuleItemInfo[]) {
+    return {
+      ...this.state.detector,
+      inputs: [
+        {
+          ...this.state.detector.inputs[0],
+          detector_input: {
+            ...this.state.detector.inputs[0].detector_input,
+            pre_packaged_rules: newRules
+              .filter((rule) => rule.enabled && rule.prePackaged)
+              .map((rule) => ({ id: rule._id })),
+            custom_rules: newRules
+              .filter((rule) => rule.enabled && !rule.prePackaged)
+              .map((rule) => ({ id: rule._id })),
```

Comment:
Not blocking for this PR, but `pre_packaged_rules` and `custom_rules` both involve looping through `newRules`. Rather than looping through `newRules`, we could take an action item to separate the `newRules` into separate variables about the return line, and then call those variables on lines 263 and 266.

Reply:
let's discuss this and make the change in a follow up PR. Thanks!